### PR TITLE
[FLORA-63] Membership data model for packages

### PR DESCRIPTION
**Pull request by mau5mat**
## Proposed changes
- [x] Query Module
  - Add functions for the following logic:
    - Query all packages belonging to a particular `packageGroupId`
    - Query a `packageGroup` by `packageGroupName`
- [x] Write Tests
  - Create a new module for the tests related to the Querys
  - Look at `PackageSpec.hs` for inspriation
  - Membership tests (removing, adding)
  - Add two new lines to `cleanUp` that will delete the new tables that have been added as part of the migrations for this ticket

## Contributor checklist

- [x] My PR is related to [FLORA-63](https://linear.app/flora-pm/issue/FLORA-63/create-a-membership-data-model-for-packages) 
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/flora-pm/flora-server/blob/development/CONTRIBUTING.md)
- [x] I have inserted my change into `changelog.d`
- [ ] I have updated documentation in `./docs/docs` if a public feature has a behaviour change

### CODE REVIEW

File: changelog.d/63 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- Created:
+    - `Flora/Model/PackageGroup/Query.hs`
+    - `Flora/Model/PackageGroup/Types.hs`
+    - `Flora/Model/PackageGroup/Update.hs`
+- Created:
+    - `Flora/Model/PackageGroupPackage/Query.hs`
+    - `Flora/Model/PackageGroupPackage/Types.hs`
+    - `Flora/Model/PackageGroupPackage/Update.hs`
+- Created:
+    - test/Flora/PackageGroupSpec.hs`
```
Comment (tchoutri) [inline changelog.d/63:18]: This part is redundant

File: src/core/Flora/Model/PackageGroup/Query.hs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+getPackagesByPackageGroupId packageGroupId = dbtToEff $ query Select q (Only packageGroupId)
+  where
+    q =
+      [sql|
+        SELECT p.*
+        FROM packages AS p
+        JOIN package_group_packages AS pgp
+        ON p.package_id = pgp.package_id
+        WHERE pgp.package_group_id = ?
+      |]
```
Comment (tchoutri) [inline src/core/Flora/Model/PackageGroup/Query.hs:33]: I think you could replace this with [joinSelectOneByField](https://hackage.haskell.org/package/pg-entity-0.0.4.4/docs/Database-PostgreSQL-Entity.html#v:joinSelectOneByField):

```haskell
joinSelectOneByField
  @Package -- `FROM`
  @PackageGroupPackage -- `INNER JOIN`
  [field| package_id |] -- `ON` field
  [field| package_group_id |] -- `WHERE` field
  (Only packageGroupId)
```

File: src/core/Flora/Model/PackageGroup/Types.hs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module Flora.Model.PackageGroup.Types where
+
+import GHC.Generics
+
+import Control.DeepSeq (NFData)
```
Comment (tchoutri) [inline src/core/Flora/Model/PackageGroup/Types.hs:5]: You can remove this empty line

**Review by tchoutri — CHANGES_REQUESTED** [review]

File: src/core/Flora/Model/PackageGroup/Query.hs (hunk 2 above)
Comment (mau5mat) [reply src/core/Flora/Model/PackageGroup/Query.hs:33]: I removed the `(Only packageGroupId)` -> `packageGroupId` here and it works, thank you :)

Comment (Kleidukos) [reply src/core/Flora/Model/PackageGroup/Query.hs:33]: Haha yes, I typed it from memory :+1:

Comment (tchoutri) [reply src/core/Flora/Model/PackageGroup/Query.hs:33]: Haha yes, I typed it from memory 👍

**Review by tchoutri — APPROVED** [review]

**Comment by tchoutri** [comment]
:tada:

**Pull request merged**